### PR TITLE
FOGL-2429: Add system test case for varying asset structure

### PR DESCRIPTION
**Pull request by Singhal-Vaibhav**
This test uses foglamp-south-http since we want to generate varying asset structure
**Usage:**
```FogLAMP/tests/system/python $ pytest e2e/test_e2e_vary_asset_http_pi.py --pi-db=<pi_db> --pi-host=<pi_host> --pi-admin=<pi_user> --pi-passwd=<pi_pwd> --pi-token="<pi_token>"```

**Result:**

```
====== test session starts ====
platform linux -- Python 3.5.3+, pytest-3.6.0, py-1.6.0, pluggy-0.6.0
rootdir: FogLAMP/tests/system/python, inifile: pytest.ini
collected 1 item                                                                                                                                                                     

TestE2EAssetHttpPI Test that data is inserted in FogLAMP and sent to PI
            start_south_north: Fixture that starts FogLAMP with south and north instance
            read_data_from_pi: Fixture to read data from PI
            Assertions:
                on endpoint GET /foglamp/asset
                on endpoint GET /foglamp/asset/<asset_name>
                data received from PI is same as data sent  .           [100%]

======== 1 passed in 46.44 seconds =====
```

### CODE REVIEW

File: tests/system/python/e2e/test_e2e_vary_asset_http_pi.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        """ Test that data is inserted in FogLAMP and sent to PI
+            start_south_north: Fixture that starts FogLAMP with south and north instance
+            read_data_from_pi: Fixture to read data from PI
+            Assertions:
+                on endpoint GET /foglamp/asset
+                on endpoint GET /foglamp/asset/<asset_name>
+                data received from PI is same as data sent"""
+
+        conn = http.client.HTTPConnection(foglamp_url)
+        time.sleep(wait_time)
```
Comment (praveen-garg) [inline tests/system/python/e2e/test_e2e_vary_asset_http_pi.py]: this wait should not be needed.

File: tests/system/python/e2e/test_e2e_vary_asset_http_pi.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+        # Send data to foglamp-south-http
+        conn_http_south = http.client.HTTPConnection("localhost:6683")
+
+        asset_name = "e2e_varying"
+        sensor_data = [{"a": 1}, {"a": 2, "b": 3}, {"b": 4}]
```
Comment (praveen-garg) [inline tests/system/python/e2e/test_e2e_vary_asset_http_pi.py]: I suggest,  we should send sensor_data again after first retrieval verification!

File: tests/system/python/e2e/test_e2e_vary_asset_http_pi.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            data = [{"asset": "{}".format(asset_name), "timestamp": "{}".format(tm), "key": str(uuid.uuid4()),
+                     "readings": d}]
+            conn_http_south.request("POST", '/sensor-reading', json.dumps(data))
+            r = conn_http_south.getresponse()
+            assert 200 == r.status
+            r = r.read().decode()
+            jdoc = json.loads(r)
+            assert {'result': 'success'} == jdoc
+
+        time.sleep(wait_time)
```
Comment (praveen-garg) [inline tests/system/python/e2e/test_e2e_vary_asset_http_pi.py]: add a new line after sleep and add a comment why sleep is needed.

File: tests/system/python/e2e/test_e2e_vary_asset_http_pi.py (hunk 2 above)
Comment (praveen-garg) [reply tests/system/python/e2e/test_e2e_vary_asset_http_pi.py]: can the next data be like:

` sensor_data = [{"b": 1.1}, {"a2": 2, "b2": 3}, {"a": 4.0}]`

Comment (Singhal-Vaibhav) [reply tests/system/python/e2e/test_e2e_vary_asset_http_pi.py]: Our current approach deletes web element from PI after reading, Instead I have added two list, 
```
        # 2 list having mixed data simulating different sensors
        # (sensors coming up and down, sensors throwing int and float data)
        sensor_data = [{"a": 1}, {"a": 2, "b": 3}, {"b": 4}]
        sensor_data_2 = [{"b": 1.1}, {"a2": 2, "b2": 3}, {"a": 4.0}]
```
which will replicate the desired scenario.

File: tests/system/python/e2e/test_e2e_vary_asset_http_pi.py (hunk 3 above)
Comment (Singhal-Vaibhav) [reply tests/system/python/e2e/test_e2e_vary_asset_http_pi.py]: done

File: tests/system/python/e2e/test_e2e_vary_asset_http_pi.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        retval = json.loads(r)
+
+        assert sensor_data_2[2] == retval[0]["reading"]
+        assert sensor_data_2[1] == retval[1]["reading"]
+        assert sensor_data_2[0] == retval[2]["reading"]
+        assert sensor_data[2] == retval[3]["reading"]
+        assert sensor_data[1] == retval[4]["reading"]
+        assert sensor_data[0] == retval[5]["reading"]
+
+        time.sleep(wait_time)
```
Comment (praveen-garg) [inline tests/system/python/e2e/test_e2e_vary_asset_http_pi.py]: add a new line after sleep and add a comment why sleep is needed.

**Pull request merged**